File: CS/Models/XPResource.cs
using System;
using DevExpress.Xpo;
using DevExpress.XtraScheduler;
using DevExpress.XtraScheduler.Xml;

public class XPResource : XPObject {
    public XPResource(Session session) : base(session) { }

    [Size(SizeAttribute.Unlimited)]  // !!! To set the Memo field type.
    public string Name;              // Resource.Caption

    [Association()]
    public XPCollection<XPAppointment> Appointments {
        get {
            return GetCollection<XPAppointment>("Appointments");
        }
    }
}

File: VB/Models/XPResource.vb
Imports Microsoft.VisualBasic
Imports System
Imports DevExpress.Xpo
Imports DevExpress.XtraScheduler
Imports DevExpress.XtraScheduler.Xml

Public Class XPResource
	Inherits XPObject
	Public Sub New(ByVal session As Session)
		MyBase.New(session)
	End Sub

	<Size(SizeAttribute.Unlimited)> _
	Public Name As String ' Resource.Caption

	<Association()> _
	Public ReadOnly Property Appointments() As XPCollection(Of XPAppointment)
		Get
			Return GetCollection(Of XPAppointment)("Appointments")
		End Get
	End Property
End Class